### ISSUE TITLE
Verhalten bei "IsActive" = false

### ISSUE THREAD

**Issue by SuperNils** [rg-engineering/ioBroker.heatingcontrol#332]
Hallo Rene,
ich verwende in meiner Visualisierung den Datenpunkt "isActive" wie folgt:
Bei true ist der Zeitplan bzw. Automatik aktiv, bei false wird auf vollständig manuellen Betrieb umgeschaltet. Dabei wird zunächst die "TemperatureIfNoHeatingPeriod" gesetzt. Anschließend kann eine beliebe Temperatur eingestellt werden.
![grafik](https://user-images.githubusercontent.com/4592954/197182107-1c1fd0f3-36a5-4920-898a-0541ecef9fc1.png)

Zu "isActive" sind mir drei Punkte aufgefallen:

1. Es scheint eine Abhängigkeit (vllt. zum Override) zu geben. Ich hatte "isActive" auf false gesetzt, die "TemperatureIfNoHeatingPeriod" wurde gesetzt, aber nach 30 min sprang der Mode auf Auto mit entsprechender Temperatur.
![grafik](https://user-images.githubusercontent.com/4592954/197182471-ce57e02e-d63f-47ff-96c1-f6451b2a0e98.png)

2. Mir würde es gefallen, wenn die Temperatur bei einem Flankenwechsel ("isActive" = false) auf den Wert des Datenpunktes "TemperatureIfNoHeatingPeriod" gesetzt wird. Anschließend wäre die Temperatur frei wählbar.

3. Wenn "isActive" auf false gesetzt ist, zeigt der Mode "no heating". In meinem Fall kann es sein, dass trotzdem geheizt wird. Wäre es möglich den Text zu ändern? Vielleicht "no schedule" oder "manual mode"?

Danke für den tollen Adapter :)

**Comment by rg-engineering** [comment]
Die Abhängigkeit zum override kann ich bestätigen: wenn override aktiv ist und man deaktviert dann den Raum, wird bei Verlassen des Overrides auf den alten Zielwert gesetzt... also als workaround immer erst den Override deaktiviern und dann den Raum deaktivieren...

**Comment by rg-engineering** [comment]
@SuperNils 
>Mir würde es gefallen, wenn die Temperatur bei einem Flankenwechsel ("isActive" = false) auf den Wert des Datenpunktes "TemperatureIfNoHeatingPeriod" gesetzt wird. Anschließend wäre die Temperatur frei wählbar.

Das tut der Adapter doch. 

![grafik](https://user-images.githubusercontent.com/8238668/205711575-b72c6f2f-6c5a-4e98-b19e-1db214ebab08.png)

dann im Datenpunkt
heatingcontrol.0.Rooms.TestRaum.TemperatureIfNoHeatingPeriod

oder 

![grafik](https://user-images.githubusercontent.com/8238668/205711654-f8c1e815-3f84-4901-baf8-0f0eee140e6f.png)

**Comment by SuperNils** [comment]
> @SuperNils
> 
> > Mir würde es gefallen, wenn die Temperatur bei einem Flankenwechsel ("isActive" = false) auf den Wert des Datenpunktes "TemperatureIfNoHeatingPeriod" gesetzt wird. Anschließend wäre die Temperatur frei wählbar.
> 
> Das tut der Adapter doch.
> 
> ![grafik](https://user-images.githubusercontent.com/8238668/205711575-b72c6f2f-6c5a-4e98-b19e-1db214ebab08.png)
> 
> dann im Datenpunkt heatingcontrol.0.Rooms.TestRaum.TemperatureIfNoHeatingPeriod
> 
> oder
> 
> ![grafik](https://user-images.githubusercontent.com/8238668/205711654-f8c1e815-3f84-4901-baf8-0f0eee140e6f.png)

Das entscheidene wäre hier die anschließend frei wählbare Temperatur. Nach einiger Zeit springt die Solltemperatur wieder auf den Wert von TemperatureIfNoHeatingPeriod. 

Bei mir läuft als workaround ein Skript welches den Datenpunkt TemperatureIfNoHeatingPeriod mit der manuell eingestellten Temperatur überschreibt. Bei einem Wechsel von "isActive = true" auf "isActive = false" wird zunächst ein fixer Wert gesetzt, anschließend wird TemperatureIfNoHeatingPeriod bei Änderung der Solltemperatur überschrieben.

Ist vielleicht zu kompliziert gedacht, aber dies ermöglicht einen komplett manuellen Betrieb des Thermostats.

**Comment by rg-engineering** [comment]
> Das entscheidene wäre hier die anschließend frei wählbare Temperatur. Nach einiger Zeit springt nämlich die Solltemperatur wieder auf den Wert von TemperatureIfNoHeatingPeriod.

ja, das kann sein. Sobald irgendetwas den Adapter triggert, schaut er in welchen state er ist und setzt die entsprechende Temperatur. 
Ich müsste also einen manuel-Modus auch für den Zustand "Heizung inaktiv" einbauen. Das gefällt mir eigentlich nicht. Warum lässt du die Heizung nicht an, änderst die Temperatur am Thermostat und nutzt somit den vorhandenen manuel-Mode?

**Comment by SuperNils** [comment]
@rg-engineering welchen vorhanden manual-mode meinst du? Es gibt vier Varianten, richtig?

**Comment by rg-engineering** [comment]
es gibt "auto" und "manual".
"auto" sagt, die Temperaturen werden laut Profil verwendet, "manual" bedeutet, der Nutzer hat eingegriffen und die Temperatur wurde manuell verändert. Dieses Handling setzt voraus, dass der Raum und die Heizperiode aktiv ist....

**Comment by SuperNils** [comment]
Mit "manual" meinst du den Override Modus, richtig? Beim Override gibt es vier Varianten.

**Comment by rg-engineering** [comment]
> Mit "manual" meinst du den Override Modus, richtig?

nein, bei override wird der Status auf "override" gesetzt. Ich meine, wenn jemand manuell z.Bsp. am Thermostat dreht und damit die Raumtemperatur verändert...
